src/ts/jsoneditor.ts

```diff
@@ -68,27 +67,36 @@ $(document).ready(function() {
           }
         }
       });
-      currentFileUploadID = $(this).data('id');
       currentFileItemID = $(this).data('uploadid');
     });
 
-    $(document).on('click', '.jsonSaver', function(){
-      if (typeof currentFileUploadID === 'undefined') {
+    // Clear the JSONEditor and unload the file
+    $(document).on('click', '.jsonClear', function() {
+      currentFileItemID = undefined;
+      editor.set({});
+      $('#jsonEditorTitle').html('File was unloaded.');
+    });
+
+    // The save function is now defined separately
+    const saveJsonFile = function(){
+      if (typeof currentFileItemID === 'undefined') {
         // we are creating a new file
         const realName = prompt('Enter name of the file');
         if (realName == null) {
           return;
         }
-        const id = $('#main_form').find('input[name="id"]').attr('value');
+        // add the new name for the file as a title
+        $('#jsonEditorTitle').html('Filename: ' + realName + '.json');
         $.post('app/controllers/EntityAjaxController.php', {
           addFromString: true,
           type: 'experiments',
-          id: id,
+          id: $('#info').data('id'),
           realName: realName,
           fileType: 'json',
           string: JSON.stringify(editor.get())
         }).done(function(json) {
-          $('#filesdiv').load('experiments.php?mode=edit&id=' + id + ' #filesdiv');
+          $('#filesdiv').load('experiments.php?mode=edit&id=' + $('#info').data('id') + ' #filesdiv');
```

Comment:
this is fine, but I found that it was better to replace with `window.location.href`. See https://github.com/elabftw/elabftw/blob/hypernext/src/ts/Link.class.ts#L36 for example.